build/int.cloudbuild.yaml

```
@@ -151,6 +151,21 @@ steps:
     - verify submodule-vpc-serverless-connector-beta
   name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
   args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && source_test_env && init_credentials && cd test/integration && RUN_STAGE=teardown go test -v ./... -p 1 -timeout 0 -run TestSubmoduleServerlessConnector']
+- id: converge private-service-connect
+  waitFor:
+    - create all
+  name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
+  args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && source_test_env && init_credentials && cd test/integration && RUN_STAGE=apply go test -v ./... -p 1 -timeout 0 -run TestPrivateServiceConnect']
```

Lets use the CFT cli instead.
```suggestion
  args: ['/bin/bash', '-c', 'cft test run TestPrivateServiceConnect --stage apply --verbose'] 
```

---

Should I change every other steps to use cft as well?

---

I mean the other tests. Maybe, I could come up with something similar to https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/build/int.cloudbuild.yaml

---

In the current code, `cft test` will not be able to find integration tests.

In order to solve this, we have two options.

1. Add the `--test-dir /workspace/test/integration` as in the foundation or
2. Add a discovery.go file as in terraform-google-cloud-storage

---

@maxi-cit lets add discovery. We can change the others in a follow up.

---

@bharathkkb this addition is running the init step in all examples, even the ones that had not been considered in the integration tests. 

This was making the build fails because some of the examples were generating version conflicts due the swapper-module-refs step. This was forcing examples to run the latest versions of the modules they were employing. Some of them having google provider ranges out of the one required by the examples. Check 'examples/routes'

I bumped the provider version on examples that were having these conflicts and it seems the building is working.